elastic/kibana
# [CCR] Auto follow pause & resume

## Summary

Addresses this issue: https://github.com/elastic/kibana/issues/48008

This contribution implements the ability to pause and resume auto-follow patterns inside of Kibana management CCR. Because to actions were added the same pattern as in rollup jobs was followed and an action menu component was created that connects to the store.

<details>
<summary>Screenshots</summary>

<img width="657" alt="Screenshot 2020-02-03 at 14 01 09" src="https://user-images.githubusercontent.com/8155004/73655919-59daca80-468f-11ea-818b-5b4f7b53f7e2.png">

<img width="323" alt="Screenshot 2020-02-03 at 14 01 24" src="https://user-images.githubusercontent.com/8155004/73655921-5a736100-468f-11ea-8497-301d64405e55.png">

<img width="503" alt="Screenshot 2020-02-03 at 14 01 28" src="https://user-images.githubusercontent.com/8155004/73655922-5a736100-468f-11ea-80f3-bba9aef3ed67.png">

<img width="367" alt="Screenshot 2020-02-03 at 14 01 37" src="https://user-images.githubusercontent.com/8155004/73655925-5a736100-468f-11ea-912f-8e101824ae40.png">

</details>

## How to test

To test this new UI most easily:

1. Set up 2 ES nodes. The first can be started with `yarn es snapshot` or `./.es/8.0.0/bin/elasticsearch`. The second should be started with `./.es/8.0.0/bin/elasticsearch -Ecluster.name="notelastic" -Ehttp.port=9201 -Epath.data=/tmp/es` to ensure that a separate cluster is created.
2. Navigate to the CCR UI and create a new Auto-follower pattern. The 2nd cluster that was just created should be discoverable at `localhost:9301` if you ran the command above.
3. Create a second auto-follower pattern so that the actions can be tested against multiple patterns.

## TODO

- [x] Fix component integration and functional tests
- [x] More manual testing required


### Release note

We surfaced this capability in the Cross Cluster Replication app.

<details>
<summary>Screenshot</summary>

<img width="1078" alt="Screenshot 2020-02-06 at 18 57 21" src="https://user-images.githubusercontent.com/8155004/73965177-f01f2280-4913-11ea-8fdc-7b9a472f7661.png">

</details>

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- ~[Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~

### For maintainers
